javascript/ql/src/semmle/javascript/StandardLibrary.qll

```diff
@@ -23,6 +23,7 @@ class CallToObjectDefineProperty extends DataFlow::MethodCallNode {
   DataFlow::Node getPropertyDescriptor() { result = getArgument(2) }
 
   /** Gets a data flow node defining a descriptor attribute of the property being defined. */
+  pragma[inline]
   DataFlow::PropWrite getAPropertyAttribute() {
```

Comment:
Can we mark this predicate is internal? Otherwise people might start using it in unexpected ways and perhaps try to do recursion through it, which won't work anymore now that it's pragma-inlined.

Reply:
That would mean we were using an internal predicate in queries, that is the kind of thing we wanted to avoid in #1211. I have reformulated the predicate in the style of `hasPropertyWrite`, the optimizer is happy with that change:

```
Project                            74e701f7ebd@5441352^  74e701f7ebd@662ad4b2c  74e701f7ebd@b5f92dfa7
A2Z-F15                            758                   740                    745
angular-gulp-webpack-starter       381                   403                    381
any-balance-providers              144                   141                    140
azure-sdk-for-node                 405                   420                    416
ChakraCore                         432                   435                    433
descartes                          247                   246                    250
gecko-dev                          1332                  1335                   1340
mxnet.js                           248                   921                    258  <<<<<
node                               761                   907                    762
tatami                             117                   71                     73
TypeScript                         51                    52                     53
```

`getAPropertyAttribute` was introduced after the feature freeze, so this is not a serious breaking change.